packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -742,6 +742,10 @@ export const GridMixin = (superClass) =>
             cell.part.add('cell', `${section}-cell`);
           }
 
+          if (column._cellPart) {
+            cell.part.add(column._cellPart);
+          }
```

Comment:
I don't really like this approach since it's kind of a new feature added just to fix the issue.
As an alternative, we could change default renderers of selection column to set `part`.

Reply:
Would an observer work? Something like `'__selectionColumnCellsChanged(_headerCell, _footerCell, _cells)',`

Reply:
If you want to avoid adding a new part name, this might be an option:
```js
this._allCells.forEach((cell) => {
  cell._content.style.overflow = 'visible';
});
```

Reply:
As discussed internally, changed to use `autoWidth` instead.